packages/webapp/pages/plus/gift/[id].tsx

```diff
@@ -0,0 +1,85 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+
+import dynamic from 'next/dynamic';
+import { useViewSize, ViewSize } from '@dailydotdev/shared/src/hooks';
+import { useRouter } from 'next/router';
+import type { GetServerSideProps } from 'next';
+import { getUserShortInfo } from '@dailydotdev/shared/src/graphql/users';
+import type { GiftUserContextData } from '@dailydotdev/shared/src/components/plus/GiftUserContext';
+import { GiftUserContext } from '@dailydotdev/shared/src/components/plus/GiftUserContext';
+import type { NextSeoProps } from 'next-seo/lib/types';
+import { getPlusLayout } from '../../../components/layouts/PlusLayout/PlusLayout';
+import { getTemplatedTitle } from '../../../components/layouts/utils';
+import { defaultOpenGraph } from '../../../next-seo';
+
+const PlusMobile = dynamic(() =>
+  import(
+    /* webpackChunkName: "plusMobile" */ '@dailydotdev/shared/src/components/plus/PlusMobile'
+  ).then((mod) => mod.PlusMobile),
+);
+const PlusDesktop = dynamic(() =>
+  import(
+    /* webpackChunkName: "plusDesktop" */ '@dailydotdev/shared/src/components/plus/PlusDesktop'
+  ).then((mod) => mod.PlusDesktop),
+);
+
+type PlusPageProps = Pick<GiftUserContextData, 'giftToUser'>;
+
+const seo: NextSeoProps = {
+  title: getTemplatedTitle('Unlock Premium Developer Features with Plus'),
+  openGraph: { ...defaultOpenGraph },
+  description:
+    'Upgrade to daily.dev Plus for an ad-free experience, custom feeds, bookmark folders, clickbait shield, and more.',
+};
+
+const PlusGiftPage = ({ giftToUser }: PlusPageProps): ReactElement => {
+  const { isReady } = useRouter();
+  const isLaptop = useViewSize(ViewSize.Laptop);
+
+  if (!isReady) {
+    return null;
+  }
+
+  return (
+    <GiftUserContext.Provider value={{ giftToUser }}>
+      {isLaptop ? <PlusDesktop /> : <PlusMobile />}
+    </GiftUserContext.Provider>
+  );
+};
+
```

Comment:
Generic comments:
- Label empty render
- Header seems like odd spacing (too much?)
- Unfinished sentence?

![Screenshot 2025-01-28 at 09 58 12](https://github.com/user-attachments/assets/e2d61cbb-d10b-4b8d-b2c1-a786c2cede3d)

What renders here?
![Screenshot 2025-01-28 at 09 59 32](https://github.com/user-attachments/assets/2988edff-6961-4f57-baf2-650a3375aeb2)

Reply:
> Label empty render
> Unfinished sentence?

Once the Paddle side is setup, these concerns should show automatically. If you try locally, the values should show.

Reply:
> Header seems like odd spacing (too much?)

That's due to the Modal's spacing to be 24px which is based from the guideline, but for some reason, on Figma the spacing is 16px.